pkgs/development/interpreters/python/hooks/editable-install-hook.sh

```diff
@@ -0,0 +1,32 @@
+editableInstallHook() {
+    echo "Executing editableInstallHook"
+    runHook preShellHook
+
+    if [ -d "${venvDir}" ]; then
+      echo "Skipping venv creation, '${venvDir}' already exists"
+      source "${venvDir}/bin/activate"
+    else
+      echo "Creating new venv environment in path: '${venvDir}'"
+      @pythonInterpreter@ -m venv "${venvDir}"
+
+      source "${venvDir}/bin/activate"
+      runHook postVenvCreation
+    fi
+
+    echo "Executing pip install -e ."
+    # use `python3` to use the python hooked up to the activated venv
+    eval "python3 -m pip install -e . --no-build-isolation >&2"
```

Comment:
1 The previous hook was using @pythonInterpreter@ ftom what I see. I think the interpreter still respected the venv when that was relevant because of the env variables set up by bin/activate. Is there an argument for changing this behaviour?
2 Are there any circumstances where the new approach could potentially result in loading the old (manylinux) libc first, thus making it impossible to use (native) dependencies from nixpkgs in the same process?

Reply:
1. It did not seem to in my testing as I recall, `@pythonInterpreter@ -m pip` tried to install to the global packages directory and failed because that is patched out because the store is read only.
2. I don't see how that would be possible, unless someone created a venv using the old libc if I understand correctly. This logic is the same as `venvShellHook`, where would the old libc come from and wouldn't that have the same problem?

Reply:
I edited the 1st message but too late:) so I'm just wondering about whether we confuse in any way the users that run nix in fhs environments

Reply:
1. That's interesting, does the wrapper reset some of the venv related env variables?

Reply:
I'm not sure there's any wrapper, I think it's just that the python3 in the store tries to install to the site packages adjacent to the store and same with the one in the venv. Activation of a venv doesn't set `PYTHONHOME`.

I don't see how there could be confusion about which python3 is being invoked, PATH will have the one in the venv first. I developed and tested all this on an FHS distro with Nix installed and it worked fine.

Reply:
Questiob: can thw interpreter in the venv be overwritten?

Reply:
to my understanding, at least in poetry, the interpreter is persisted as a symlink at `$VIRTUAL_ENV/bin/python`

Reply:
Yes, that path is a symlink to `@pythonInterpreter@`.

This does compose properly with other Python packages too in shells.

Reply:
But what would happen after some time passes and the python interpreter in the nix store is garbage collected in favor of a new one?

Reply:
Then you will have to delete the venv and let it be re-created. I am somewhat ambivalent about this solution of using a venv but it does seem better than allowing `$TEMP` to be filled up with dozens of random installs like `setuptoolsShellHook`. That does get cleaned up semi-regularly by the system though.

Do you think we should switch back to using a temporary directory? That would solve all these concerns here.

Reply:
`python -m venv` lists the following option:
```
  --upgrade             Upgrade the environment directory to use this version of Python, assuming Python has been upgraded in-place.
```

Reply:
This wouldn't end up working because the original python interpreter would be gone.